centreon-certified/elasticsearch/elastic-events-apiv2.lua

```diff
@@ -0,0 +1,294 @@
+
+function EventQueue.new(params)
+    local self = {}
+
+    local mandatory_parameters = {
+        "elastic_url",
+        "elastic_token",
+        "elastic_index_status"
+    }
+
+    self.fail = false
+
+    -- set up log configuration
+    local logfile = params.logfile or "/var/log/centreon-broker/elastic-events-apiv2.log"
+    local log_level = params.log_level or 1
+
+    -- initiate mandatory objects
+    self.sc_logger = sc_logger.new(logfile, log_level)
+    self.sc_common = sc_common.new(self.sc_logger)
+    self.sc_broker = sc_broker.new(self.sc_logger)
+    self.sc_params = sc_params.new(self.sc_common, self.sc_logger)
+
+    -- checking mandatory parameters and setting a fail flag
+    if not self.sc_params:is_mandatory_config_set(mandatory_parameters, params) then
+      self.fail = true
+    end
+
+    -- overriding default parameters for this stream connector if the default values doesn't suit the basic needs
+    self.sc_params.params.accepted_categories = params.accepted_categories or "neb"
+    self.sc_params.params.accepted_elements = params.accepted_elements or "host_status,service_status"
+
+    -- apply users params and check syntax of standard ones
+    self.sc_params:param_override(params)
+    self.sc_params:check_params()
+
+    self.sc_macros = sc_macros.new(self.sc_params.params, self.sc_logger)
+    self.format_template = self.sc_params:load_event_format_file()
+    self.sc_params:build_accepted_elements_info()
+    self.sc_flush = sc_flush.new(self.sc_params.params, self.sc_logger)
+
+    local categories = self.sc_params.params.bbdo.categories
+    local elements = self.sc_params.params.bbdo.elements
+
+    self.format_event = {
+      [categories.neb.id] = {
+        [elements.host_status.id] = function () return self:format_event_host() end,
+        [elements.service_status.id] = function () return self:format_event_service() end
+      },
+      [categories.bam.id] = {}
+    }
+
+    self.send_data_method = {
+      [1] = function (data, element) return self:send_data(data, element) end
+    }
+
+    -- return EventQueue object
+    setmetatable(self, { __index = EventQueue })
+    return self
+  end
+
+--------------------------------------------------------------------------------
+---- EventQueue:format_event method
+---------------------------------------------------------------------------------
+function EventQueue:format_accepted_event()
+    local category = self.sc_event.event.category
+    local element = self.sc_event.event.element
+    local template = self.sc_params.params.format_template[category][element]
+    self.sc_logger:debug("[EventQueue:format_event]: starting format event")
+    self.sc_event.event.formated_event = {}
+
+    if self.format_template and template ~= nil and template ~= "" then
+      for index, value in pairs(template) do
+        self.sc_event.event.formated_event[index] = self.sc_macros:replace_sc_macro(value, self.sc_event.event)
+      end
+    else
```

Comment:
this is going to be deprecated in the next lib release. 

it will become 

```lua
if self.format_template and template ~= nil and template ~= "" then
  self.sc_event.event.formated_event = self.sc_macros:replace_sc_macro(template, self.sc_event.event)
else
```

`template` will no longer be a table but a json string. Macros will be replaced in the string and then the string will be decoded to return a lua table. 
this will allow multi level json

Reply:
nice. done.